### PR TITLE
docs: add in-situ analysis

### PR DESCRIPTION
**Pull request by BenWibking**
### Description
Adds documentation for enabling *in-situ* analysis of simulations.

Also: updates the documentation on how to read a time series of plotfiles with VisIt.

### Related issues
N/A

### Checklist
_Before this pull request can be reviewed, all of these tasks should be completed. Denote completed tasks with an `x` inside the square brackets `[ ]` in the Markdown source below:_
- [x] I have added a description (see above).
- [x] I have added a link to any related issues see (see above).
- [x] I have read the [Contributing Guide](https://github.com/quokka-astro/quokka/blob/development/CONTRIBUTING.md).
- [ ] I have added tests for any new physics that this PR adds to the code.
- [ ] I have tested this PR on my local computer and all tests pass.
- [ ] I have manually triggered the GPU tests with the magic comment `/azp run`.
- [x] I have requested a reviewer for this PR.

### CODE REVIEW

**Comment by markkrumholz** [comment]
I'll hold off on reviewing this until #581 is finalized and merged, since the exact functionality is still being iterated.

File: docs/insitu_analysis.rst
Comment (markkrumholz) [inline docs/insitu_analysis.rst]: Line 83 should be eliminated, based on the changes made to #581.

File: docs/insitu_analysis.rst
Comment (markkrumholz) [inline docs/insitu_analysis.rst]: You should be specific on what is computed for each of the options. That is, this should say specifically that the quantity returned is the total number of cells, total volume, or total mass in each bin, excluding cells that are masked by higher-level data.

File: docs/insitu_analysis.rst
Comment (markkrumholz) [inline docs/insitu_analysis.rst]: I would not use the word weighting, because this implies you are doing some sort of normalization. I would say histograms by mass, volume, or cell-count.

**Review by markkrumholz — COMMENTED** [review]
See inline comments.

**Comment by BenWibking** [comment]
I've re-worded things to avoid "weighting" as you suggested. Let me know if it is sufficiently clear.

**Comment by BenWibking** [comment]
@markkrumholz Can you re-review?

**Review by markkrumholz — COMMENTED** [review]
Two final minor comments: in the example input file for on-the-fly PDFs, the documentation / comments do not currently state the default behavior for the weight-by and min/max options. You should state what these default to, i.e., if you don't specify how to weight, is it weighted by mass, volume, cell count? What are the default choices for min/max if you don't specify?

File: docs/insitu_analysis.rst
Comment (BenWibking) [inline docs/insitu_analysis.rst:80]: Default is "volume".

File: docs/insitu_analysis.rst
Comment (BenWibking) [inline docs/insitu_analysis.rst:82]: Explained the defaults for min/max bins here.

**Comment by BenWibking** [comment]
> Two final minor comments: in the example input file for on-the-fly PDFs, the documentation / comments do not currently state the default behavior for the weight-by and min/max options. You should state what these default to, i.e., if you don't specify how to weight, is it weighted by mass, volume, cell count? What are the default choices for min/max if you don't specify?

I've added both of these. See inline comments.

**Review by markkrumholz — APPROVED** [review]

**Pull request merged**
